### DISCUSSION TITLE
fspicker lag on heavy directories

### DISCUSSION THREAD

**Discussion by KRRT7** [davep/textual-fspicker#5] [Q&A]
all of Fspicker methods lag when navigating to a heavy (disk space) directory, perhaps this can be improved?

**Comment by davep** [comment]
When you say the "Fspicker methods lag", what do you mean exactly? The way the library is designed is that a very large directory might take some time to load up and display, but any other code running should still be fine; the core loading code uses a threaded worker.

What it doesn't do however is trickle the directory entries into the display; it will load up the data and once everything is loaded then display it. The main purpose behind this is that all of the directory entries are needed before the sorting method can be applied.

If I get some time I could possibly look at allowing a gradual display of entries and then perform a final sort; but then that could be an annoying experience for the end user as the display appears to change while they're interacting with it.

But put some of this into context: would it be possible to get an idea of the environment this is happening in, the OS and filesystem, the size of the directory in terms of entries, etc?
